### PR TITLE
Milestone 4 Book store project

### PR DESCRIPTION
**Pull request by imhilla**
Hello TSE, 
Please review my fourth milestone project of the redux react section.
-Added filter component, actions, and reducers
Thank you,
Hillary

### CODE REVIEW

File: src/reducers/filter.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable react/no-unused-state */
+/* eslint-disable react/prefer-stateless-function */
+const filter = initialState => (state = initialState, action) => {
```
Comment (mosaaleb) [inline src/reducers/filter.js:1]: - [ ] Why disabling these eslint rules? I think we don't need it.

**Review by mosaaleb — CHANGES_REQUESTED** [review]
### Changes required

Well done @imhilla :clap:. Most of the requirements are handled, yet we have some work to do before moving on.


- [ ] Add a short summary of this pull request in the description.


- [ ] Let's start working on the Readme file.


- [ ] There are some errors in the console, let's fix them

Screenshot from 2020-10-23 10-56-18

Use [`shape`](https://reactjs.org/docs/typechecking-with-proptypes.html) to validate an object with a certain shape.

Please handle these requirements and submit a new code review.

##### Muhammad Ebeid

File: src/reducers/books.js [hunk 2]
```diff
@@ -1,7 +1,11 @@
 /* eslint-disable react/no-unused-state */
 /* eslint-disable react/prefer-stateless-function */
```
Comment (bellom) [inline src/reducers/books.js]: we are indirectly changing the rules by disabling these rules, let's fix the codebase instead.

File: src/containers/booksform.js [hunk 3]
```diff
@@ -30,53 +30,42 @@ class BooksForm extends React.Component {
...
-                </form>
-              </div>
+      <div className="addContainer">
+        <div>
+          <h5 className="titlee">ADD NEW BOOK</h5>
+        </div>
+        <div>
+          <form id="bookForm" name="updateTaskForm" className="formContainer">
+            <div className="inputw">
+              {/* eslint-disable-next-line no-return-assign */}
```
Comment (bellom) [inline src/containers/booksform.js]: we are indirectly changing the rules by disabling these rules, let's fix the codebase instead.

File: src/containers/booksform.js [hunk 4]
```diff
@@ -30,53 +30,42 @@ class BooksForm extends React.Component {
 
   render() {
     const stringId = 'category';
-    const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+    const categories = ['Choose a category', 'Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
     const renderCategories = categories.map((item, id) => (
       // eslint-disable-next-line no-return-assign
```
Comment (bellom) [inline src/containers/booksform.js]: we are indirectly changing the rules by disabling these rules, let's fix the codebase instead.

File: src/containers/booklist.js [hunk 5]
```diff
@@ -2,52 +2,68 @@ import React from 'react';
...
 
 const mapStateToProps = state => ({
-  books: state.books,
+  books: state.books.books,
+  category: state.books.filter,
 });
 
 BookList.propTypes = {
-  books: PropTypes.objectOf,
+  books: PropTypes.arrayOf(PropTypes.any),
```
Comment (bellom) [inline src/containers/booklist.js:51]: According to the previous @mosaaleb told you to use `props-types.shape`

Using [PropTypes.shape](https://dev.to/cesareferrari/how-to-specify-the-shape-of-an-object-with-proptypes-3c56) here will allow you to set a type check that is more accurate, then you can remove the inline comment to disable prop-types.

File: src/containers/booksform.js [hunk 6]
```diff
@@ -30,53 +30,42 @@ class BooksForm extends React.Component {
...
+              {/* eslint-disable-next-line no-return-assign */}
+              <input type="text" id="fname" name="title" ref={node => (this.inputNode = node)} className="inputLesson" />
+            </div>
+            <div className="dropdown">
+              <select
+                className="custom-select"
+                id="inputGroupSelect01"
+                onChange={this.handleDrop}
+                ref={this.myRef}
+                // eslint-disable-next-line react/destructuring-assignment
```
Comment (bellom) [inline src/containers/booksform.js]: we are indirectly changing the rules by disabling these rules, let's fix the codebase instead.

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @imhilla   :clap: 

**Great work making `almost` all the necessary corrections from the previous review.**

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by mosaaleb — CHANGES_REQUESTED** [review]
#### Changes Required :heavy_multiplication_x: 

Thanks for working on the previous review requirements. Still, I got this error when I tried to add multiple books.


Screenshot from 2020-10-25 17-57-02


It seems that that given `id` for the new books are the same. Please fix this issue and submit a new code review.


##### Muhammad Ebeid

**Review by jofer86 — APPROVED** [review]
## `const PROJECT_STATUS = "APPROVED"` 🚀 🚀 🚀 
## :wave: :wave: Hello! I'm Jorge, and  I will be conducting this code review. Let me get a closer look 🔎 🔍 🔎 🔍 🕵🏼 

- All of the required changes were taken care of. Great Job!!!! you might move along.

--------------------------------------------------------------------
**Rock on**  🤘🏼 
**_Jorge Fernando Rincón Figueroa._** 
Technical Support Engineer.

**Pull request merged**